![Data-Preparation.jpg](https://images.xenonstack.com/blog/Data-Preparation.jpg " This is data prepration Step")

# Titanic Exploratory Data Analysis

## Table of Contents

1. [Problem Statement](#section1)<br>
2. [DataSet Loading and Brif Description](#section2)<br>
    - 2.1 [Variable Notes](#section201)<br>
3. [Examining Data Available From An Existing Information Source-Data profiling](#section3)<br>
    - 3.2 [Basic DataSet Understanding](#section301)<br>

### 1. Problem Statement


__What sorts of people were likely to survive.__

We will be doing and exploring with help of basic uses of __numpy__ and __pandas__ and will cover basic syntex/command of __Exploratory Data Analysis(EDA)__  which includes __cleaning__, __munging__, __combining__, __reshaping__, __slicing__, __dicing__, and __transforming data__ for analysis purpose.

- Defining the problem statement (Intrduction)
- Collecting the data (Source)
- Preprocessing the data (Data wrangling)
- Exploratory data analysis (EDA)<br>
    Understand the data by EDA and EDA is a critical and first step in analyzing the data.
    - Finding patterns in Data ( Classification, Clustering, Correlation, Regression .. )
    - Checking of assumptions
    - Determining relationships in Data
    - Preliminary selection of appropriate technology  ( use of techniques of system analysis/operations research in the selection of the optimal - or most appropriate)
    - Detection of mistakes


### 2. DataSet Loading and Brif Description

| Column Name   | Description                                               |
| ------------- |-------------                                              | 
| PassengerId   | Passenger Identity                                        | 
| Cabin         | Cabin number                                              |
| Name          | Name of passenger                                         |   
| Sex           | Sex of passenger                                          |
| Age           | Age of passenger                                          |
| Pclass        | Class of ticket  (1 = first; 2 = second; 3 = third)       | 
| SibSp         | Number of sibling and/or spouse travelling with passenger |
| Parch         | Number of parent and/or children travelling with passenger|
| Ticket        | Ticket number                                             |
| Fare          | Price of ticket                                           |
| Embarked      | Port of embarkation (C = Cherbourg; Q = Queenstown; S = Southampton)|
| Survived      | Whether passenger survived or not Survival (0 = no; 1 = yes)| 

<a id=section201></a>

### 2.1 Variable Notes
__pclass:__<br> A proxy for socio-economic status (SES)
- 1st = Upper
- 2nd = Middle
- 3rd = Lower

__age:__ <br> Age is fractional if less than 1. If the age is estimated, is it in the form of xx.5

__sibsp:__ The dataset defines family relations in this way...
- Sibling = brother, sister, stepbrother, stepsister
- Spouse = husband, wife (mistresses and fiancés were ignored)

__parch:__ The dataset defines family relations in this way...
- Parent = mother, father
- Child = daughter, son, stepdaughter, stepson
- Some children travelled only with a nanny, therefore parch=0 for them.

In [18]:
# Importing packges

import numpy as np              # milti-dimensional array and matrices
import pandas as pd             # manipulation and analysis
import pandas_profiling 
import matplotlib.pyplot as plt # umerical mathematics extension NumPy ( Ploting libarary)
import seaborn as sns           # Provides a high level interface for drawing attractive and informative statistical graphics

'''
%matplotlib inline sets the backend of matplotlib to the 'inline' backend: With this backend, the output of 
plotting commands is displayed inline within frontends like the Jupyter notebook, directly below the code cell that
produced it.
'''

%matplotlib inline

"\n%matplotlib inline sets the backend of matplotlib to the 'inline' backend: With this backend, the output of \nplotting commands is displayed inline within frontends like the Jupyter notebook, directly below the code cell that\nproduced it.\n"

In [26]:
# we can see the value of multiple statements at once
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"

# Importing data Set Desktop/TanujGit/ExploratoryDataAnalysis-EDA/DataSet

TitanicData = pd.read_csv("C://Users//t.shah//Desktop//TanujGit//ExploratoryDataAnalysis-EDA//DataSet/Titanic_DataSet.csv")
TitanicData.head()
TitanicData.tail()

PassengerId Cabin                                               Name  \
0            1   NaN                            Braund, Mr. Owen Harris   
1            2   C85  Cumings, Mrs. John Bradley (Florence Briggs Th...   
2            3   NaN                             Heikkinen, Miss. Laina   
3            4  C123       Futrelle, Mrs. Jacques Heath (Lily May Peel)   
4            5   NaN                           Allen, Mr. William Henry   

      Sex   Age  Pclass  SibSp  Parch            Ticket     Fare Embarked  \
0    male  22.0       3      1      0         A/5 21171   7.2500        S   
1  female  38.0       1      1      0          PC 17599  71.2833        C   
2  female  26.0       3      0      0  STON/O2. 3101282   7.9250        S   
3  female  35.0       1      1      0            113803  53.1000        S   
4    male  35.0       3      0      0            373450   8.0500        S   

   Survived  
0         0  
1         1  
2         1  
3         1  
4         0

PassengerId Cabin                                      Name     Sex  \
886          887   NaN                     Montvila, Rev. Juozas    male   
887          888   B42              Graham, Miss. Margaret Edith  female   
888          889   NaN  Johnston, Miss. Catherine Helen "Carrie"  female   
889          890  C148                     Behr, Mr. Karl Howell    male   
890          891   NaN                       Dooley, Mr. Patrick    male   

      Age  Pclass  SibSp  Parch      Ticket   Fare Embarked  Survived  
886  27.0       2      0      0      211536  13.00        S         0  
887  19.0       1      0      0      112053  30.00        S         1  
888   NaN       3      1      2  W./C. 6607  23.45        S         0  
889  26.0       1      0      0      111369  30.00        C         1  
890  32.0       3      0      0      370376   7.75        Q         0

<a id = section3> </a>

### 3.Examining Data Available From An Existing Information Source-Data profiling

<a id=section301> </a>

### 3.1 Basic DataSet Understanding

In [28]:
TitanicData.shape

(891, 12)

In [29]:
TitanicData.columns

Index(['PassengerId', 'Cabin', 'Name', 'Sex', 'Age', 'Pclass', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Embarked', 'Survived'],
      dtype='object')

In [30]:
TitanicData.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
PassengerId    891 non-null int64
Cabin          204 non-null object
Name           891 non-null object
Sex            891 non-null object
Age            714 non-null float64
Pclass         891 non-null int64
SibSp          891 non-null int64
Parch          891 non-null int64
Ticket         891 non-null object
Fare           891 non-null float64
Embarked       889 non-null object
Survived       891 non-null int64
dtypes: float64(2), int64(5), object(5)
memory usage: 83.6+ KB


#### Observation:


- Total row __891__ and columns are __12__
- PassengerId, Name, Sex, Pclass, sibsp, Parch, Ticket, Fare and Survived, which has all 891 has __NOT Null__ values.
- Cabin  __78%__ Null values
- Age __20% Null__ Values
- Embarked   __~0.03%__ Null Values ( only 2 records has null values)


In [34]:
TitanicData.isnull().sum()

PassengerId      0
Cabin          687
Name             0
Sex              0
Age            177
Pclass           0
SibSp            0
Parch            0
Ticket           0
Fare             0
Embarked         2
Survived         0
dtype: int64

In [31]:
TitanicData.describe()
# include='all' gives the summary statistics of all the columns.

PassengerId         Age      Pclass       SibSp       Parch  \
count   891.000000  714.000000  891.000000  891.000000  891.000000   
mean    446.000000   29.699118    2.308642    0.523008    0.381594   
std     257.353842   14.526497    0.836071    1.102743    0.806057   
min       1.000000    0.420000    1.000000    0.000000    0.000000   
25%     223.500000   20.125000    2.000000    0.000000    0.000000   
50%     446.000000   28.000000    3.000000    0.000000    0.000000   
75%     668.500000   38.000000    3.000000    1.000000    0.000000   
max     891.000000   80.000000    3.000000    8.000000    6.000000   

             Fare    Survived  
count  891.000000  891.000000  
mean    32.204208    0.383838  
std     49.693429    0.486592  
min      0.000000    0.000000  
25%      7.910400    0.000000  
50%     14.454200    0.000000  
75%     31.000000    1.000000  
max    512.329200    1.000000

#### Observation :
- The mean, min and max are self-explanatory.
- std is the standard deviation, which shows how data is dispersed.
- The larger Variance /Standard Deviation in Dataset more dispersed in dataset. You can see STD value is 49.693429 for FARE. And min is zero and max value 512.32920.
- Function gives the __mean, std and IQR(Interquartile range)__ values. It excludes character column and calculate summary statistics only for numeric columns
- Min “__Fare__“ is zero. There might be politics, coupon , staff, pilot seat. There could be any reason whose fare is ZERO.
- For __Pclass__, mean (2.308642) is lower than the median (3.0000)(50%). This shows that the classes of the passenger was skewed more towards 3rd class, as most people were on the 3rd class.
- For Fare, mean (32.204208) is higher than the median (14.454200). This shows that the fares were skewed more towards lower fares, as most people were on the 3rd class
- For __SibSp__, the mean is (0.523008) is higher than the median(0.0000). The skew is to the left as many passengers were travelling without a spouse or sibling.
- For __Survived__, mean (0.383838) is higher than the median (0.0000). This is because many passengers died, thus skew was to the left.


In [ ]:
TitanicProfile = pandas_profiling.ProfileReport(TitanicData)
TitanicProfile.to_file(outputfile="titanic_before_preprocessing.html")

Exception in thread Thread-8:
Traceback (most recent call last):
  File "C:\Users\t.shah\AppData\Local\Continuum\anaconda3\lib\threading.py", line 916, in _bootstrap_inner
    self.run()
  File "C:\Users\t.shah\AppData\Local\Continuum\anaconda3\lib\threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\t.shah\AppData\Local\Continuum\anaconda3\lib\multiprocessing\pool.py", line 463, in _handle_results
    task = get()
  File "C:\Users\t.shah\AppData\Local\Continuum\anaconda3\lib\multiprocessing\connection.py", line 251, in recv
    return _ForkingPickler.loads(buf.getbuffer())
ModuleNotFoundError: No module named 'pandas.core.internals.managers'; 'pandas.core.internals' is not a package

